[jerrelllee] iP

jerrelllee opened the pull request:
# Fence
It's: 
- text-based
-  easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need is:
1. download it from [here](https://github.com/jerrelllee/ip/releases)
2. double-click it
3. add your tasks
4. let it manage your tasks for you :wink:

> [!IMPORTANT]
> Some of the features are still being developed.

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jerrelllee committed:
Renamed chatbot and added greeting

jerrelllee committed:
Added echo, make Fence customised and more like an object

jerrelllee committed:
Replaced echo with add, added list

jerrelllee committed:
Added task class, added mark/unmark and further customised Fence

jerrelllee committed:
Added 3 types of tasks, changed logic to support new tasks

jerrelllee committed:
Added Text UI testing

jerrelllee committed:
Handled possible errors from incorrect inputs

jerrelllee committed:
Added delete

jerrelllee committed:
Add save

jerrelllee committed:
Merge branch 'branch-Level-7'

jerrelllee committed:
Add time and date format for deadlines

jerrelllee committed:
Merge branch 'branch-Level-8'

jerrelllee committed:
Add parser, storage, tasklist and ui classes

jerrelllee committed:
Merge branch 'branch-A-MoreOOP'

jerrelllee committed:
Add fence, parser, storage, task, tasklist and ui packages

jerrelllee committed:
Merge branch 'branch-A-Packages'

jerrelllee committed:
Merge branch 'add-gradle-support'

jerrelllee committed:
Update build.gradle for fence

jerrelllee committed:
Improve exception handling and rename methods clearer

jerrelllee committed:
Add JUnit tests

jerrelllee committed:
Merge branch 'branch-A-JUnit'

jerrelllee committed:
Add JavaDocs for Ui and TaskList class

jerrelllee committed:
Follow naming CodingStandard for TaskList and Ui classes

dominic2412 commented on src/main/java/fence/Fence.java:
```diff
@@ -0,0 +1,86 @@
+package fence;
```
Your error handling with custom exceptions (e.g., NoSuchElementException, DateTimeParseException) is effective.
Consider adding more specific error messages to guide the user on what they did wrong. This can enhance the user experience.

dominic2412 commented on src/main/java/fence/parser/Parser.java:
```diff
@@ -0,0 +1,177 @@
+import fence.task.Todo;
+import java.time.LocalDate;
+import java.util.NoSuchElementException;
+import java.util.StringTokenizer;
```
The use of StringTokenizer to parse commands is a clever choice. However, it might be a bit hard to follow for those who are not familiar with StringTokenizer.

dominic2412 commented on src/main/java/fence/task/Task.java:
```diff
@@ -0,0 +1,28 @@
+package fence.task;
```
The task management logic is clean and straightforward.

dominic2412 commented on src/main/java/fence/ui/Ui.java:
```diff
@@ -0,0 +1,66 @@
+package fence.ui;
```
The UI responses in the Ui class are user-friendly and straightforward, goodjob!

dominic2412 reviewed (commented):
I really like how you've approached this task, especially the clear and logical flow of the Fence class. With a few tweaks and improvements, this could be even better! Keep up the good work, and I'm looking forward to seeing how you continue to refine this project.

jerrelllee committed:
Add find

jerrelllee committed:
Add JavaDocs for task classes

jerrelllee committed:
Follow CodingStandard for remaining classes

jerrelllee committed:
Merge branch 'branch-A-JavaDoc'

jerrelllee committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/fence/tasklist/TaskList.java
#	src/main/java/fence/ui/Ui.java

Resolution: keep both

jerrelllee committed:
Merge branch 'branch-Level-9'

sayglenn commented on src/main/java/fence/storage/Storage.java:
```diff
@@ -0,0 +1,59 @@
+        }
+    }
+
+    public ArrayList<Task> read(Parser parser) {
```
Perhaps this could be named better? e.g. readTasks

sayglenn commented on src/main/java/fence/parser/Parser.java:
There's no javadocs in this file, which could slightly be confusing since it may not be easy discerning between two similarly named functions `ParseTxt` and `ParseInput`. Perhaps javadocs to discern the two would be suitable?

sayglenn commented on src/main/java/fence/task/Task.java:
```diff
@@ -0,0 +1,55 @@
+    /**
+     * Sets the isDone field as true.
+     */
+    public void complete() {
```
Perhaps the method name could be more specific on what is being completed?

sayglenn commented on src/main/java/fence/task/Task.java:
```diff
@@ -0,0 +1,55 @@
+    /**
+     * Sets the isDone field as false.
+     */
+    public void undo() {
```
Similarly as for the complete method above, could the name be more specific?

sayglenn commented on src/test/java/fence/parser/ParserTest.java:
The naming conventions for your tests are well done! I could understand the purposes of these tests clearly.

sayglenn commented on src/main/java/fence/ui/Ui.java:
```diff
@@ -0,0 +1,115 @@
+     * Prints out the size of the task list.
+     * @param size Size of the task list.
+     */
+    public void count(int size) {
```
This could be more specific, perhaps countTasks will be more suitable?

sayglenn reviewed (commented):
Some javadocs are missing for some classes which I assume would be added in due time. However, the general flow of the code looks good! I especially liked the descriptive titles of your test cases. Good job!

muller317 commented on src/main/java/fence/tasklist/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+     * Appends the given task to the end of the list.
+     * @param task Task to be added.
+     */
+    public void add(Task task) {
```
Would it be better to name it as addTask instead?

muller317 reviewed (commented):
Overall the code adhere to the coding standard well, the OOP structure is clear and well-organized.

jerrelllee committed:
Add checkstyle in gradle

jerrelllee committed:
Merge branch 'branch-A-CheckStyle'

jerrelllee committed:
Replace UI with GUI

jerrelllee committed:
Merge branch 'branch-Level-10'

jerrelllee committed:
Add assertions

jerrelllee committed:
Improve code quality

getResponse method: uses low level methods to handle each command.

getResponse method is too long and does not have SLAP.

Let's abstract out the respective handling of each command into new
methods.

Abstracting out the handling of each command ensures highest level
of abstraction possible and makes it more readable.

jerrelllee committed:
Improve code quality of ParserTest class

Methods whose test cases are expected to throw exceptions rely on
empty catch blocks to pass the test case.

Empty catch block may confuse readers.

Let's add comments explaining why the catch block is left empty.

Adding comments explains that the exceptions are not just being
ignored and this is the expected behavior.

jerrelllee committed:
Improve code quality of Parser class

parseTxt and parseInput methods use low level methods to parse
each command.

parseTxt and parseInput methods are too long and do not have SLAP.

Let's abstract out the respective handling of each command into new
methods.

Abstracting out the handling of each command ensures highest level
of abstraction possible and makes it more readable.

jerrelllee committed:
Improve code quality

jerrelllee committed:
Merge pull request #2 from jerrelllee/branch-A-Assertions

Add assertions

jerrelllee committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/fence/Fence.java

Resolution: keep both

jerrelllee committed:
Merge pull request #3 from jerrelllee/branch-A-CodeQuality

Improve code quality

jerrelllee committed:
Add automatic reminders for deadline tasks

jerrelllee committed:
Merge branch 'branch-B-Reminders'

jerrelllee committed:
Improve general functionalities and JavaDocs

jerrelllee committed:
Update README.md

jerrelllee committed:
Replace assertion with exception when parsing storage

jerrelllee committed:
Update README.md

jerrelllee committed:
Update README.md

jerrelllee committed:
Update README.md

jerrelllee committed:
Update README.md
